src/main/scala/hellotest/FunctionalWordCloud.scala

```diff
@@ -1,15 +1,29 @@
 package hellotest
 
+import scala.io.Source
+
 def readIgnoreFile(ignoreFilePath: String): Set[String] = {
   // DO: move the code for reading the ignore file content into function.
   // Also, close source and remove redundant toString
-  Set.empty
+  if (ignoreFilePath.nonEmpty) {
+    import scala.language.unsafeNulls
+    val file = Source.fromFile(ignoreFilePath);
```

Comment:
You're right about adding error handling. You can use the util.Try for a more functional approach. Here's the [link](https://www.scala-lang.org/api/current/scala/util/Try.html) to the API.